app/src/main/java/com/google/maps/android/compose/MapSampleActivity.kt

```diff
@@ -90,10 +93,15 @@ class MapSampleActivity : ComponentActivity() {
 @Composable
 private fun GoogleMapView(modifier: Modifier, onMapLoaded: () -> Unit) {
     val singapore = LatLng(1.35, 103.87)
+    var markerPosition by remember { mutableStateOf(singapore) }
     // Observing and controlling the camera's state can be done with a CameraPositionState
     val cameraPositionState = rememberCameraPositionState {
         position = CameraPosition.fromLatLngZoom(singapore, 11f)
     }
+    val markerDragState = rememberMarkerDragState()
+    if (markerDragState.dragState == DragState.END) {
+        markerDragState.marker.value?.let { markerPosition = it.position }
```

Comment:
Hm this is an interesting consequence of this feature. The initially set position state will no longer accurate when dragged which is already true for the existing version of the library. Having to do the above is not ideal, the `Marker` composable should handle this for you.

I think a more intuitive design would be to emulate how `CameraPositionState` works. So something like the following:

```kotlin
class MarkerPositionState(
    position: LatLng = LatLng(0.0f, 0.0f)
) {
    var position: LatLng by mutableStateOf(position)

    var dragState: DragState by mutableStateOf(DragState.END)
        internal set
}
```

This means updating the `position` property to accept `MarkerPositionState` and removal of the `markerDragState` as the `DragState` would now be accessed via `MarkerPositionState`.

Some pros/cons of this:
  * pro: `position` would always be consistent as the marker is dragged
  * con: would cause a breaking change
  * con: may have some odd behaviors when the same state is passed to multiple marker instances

Other considerations:
  * note that only the `position` is exposed and not the full `Marker` object. I'm not convinced that it should be exposed unless it is useful for the `tag` property to be accessed while the `Marker` is dragged.